### PR TITLE
Marking .abs() method for floating point types as obsolete in Swift 4

### PR DESCRIPTION
**Pull request by moiseev**
In the name of source compatibility, the `abs()` static methods return to all floating point types.

Resolves <rdar://problem/28456614>.

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by jrose-apple** [comment]
You can still mark them as deprecated in Swift 3.1 as well!

**Comment by moiseev** [comment]
@swift-ci Please smoke test Linux platform

**Comment by moiseev** [comment]
Updated.
@swift-ci Please smoke test and merge

File: stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb [hunk 1]
```diff
@@ -336,9 +336,11 @@ extension CGFloat {
     fatalError("unavailable")
   }
 
-  @available(*, unavailable, message: "Please use the `abs(_:)` free function")
+  @available(swift, deprecated: 3.1, message: "Please use the `abs(_:)` free function")
+  @available(swift, obsoleted: 4.0, message: "Please use the `abs(_:)` free function")
```
Comment (rintaro) [inline stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb]:
```swift
@available(swift, deprecated: 3.1, obsoleted: 4.0, message: "...")
```
doesn't work?

Comment (moiseev) [inline stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb]: Good question. Now that you've mentioned it, I think it should.

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

**Comment by moiseev** [comment]
@swift-ci Please smoke test and merge

**Comment by moiseev** [comment]
Unrelated swiftpm test failure.
@swift-ci Please smoke test Linux platform

**Comment by moiseev** [comment]
@swift-ci Please smoke test Linux platform

**Pull request merged**